components/provider_ppucd/src/structs.rs

```diff
@@ -0,0 +1,156 @@
+    pub const EPRES_V1: DataKey = data_key!(uniset, "EPres", 1);
+    pub const EXT_V1: DataKey = data_key!(uniset, "Ext", 1);
+    pub const EXT_PICT_V1: DataKey = data_key!(uniset, "ExtPict", 1);
+    pub const GRAPH_V1: DataKey = data_key!(uniset, "graph", 1);
+    pub const GR_BASE_V1: DataKey = data_key!(uniset, "Gr_Base", 1);
+    pub const GR_EXT_V1: DataKey = data_key!(uniset, "Gr_Ext", 1);
+    pub const GR_LINK_V1: DataKey = data_key!(uniset, "Gr_Link", 1);
+    pub const HEX_V1: DataKey = data_key!(uniset, "Hex", 1);
+    pub const HYPHEN_V1: DataKey = data_key!(uniset, "Hyphen", 1);
+    pub const IDC_V1: DataKey = data_key!(uniset, "IDC", 1);
+    pub const IDEO_V1: DataKey = data_key!(uniset, "Ideo", 1);
+    pub const IDS_V1: DataKey = data_key!(uniset, "IDS", 1);
+    pub const IDSB_V1: DataKey = data_key!(uniset, "IDSB", 1);
+    pub const IDST_V1: DataKey = data_key!(uniset, "IDST", 1);
+    pub const JOIN_C_V1: DataKey = data_key!(uniset, "Join_C", 1);
+    pub const LOE_V1: DataKey = data_key!(uniset, "LOE", 1);
+    pub const LOWER_V1: DataKey = data_key!(uniset, "Lower", 1);
+    pub const MATH_V1: DataKey = data_key!(uniset, "Math", 1);
+    pub const NCHAR_V1: DataKey = data_key!(uniset, "NChar", 1);
+    pub const NFCINERT_V1: DataKey = data_key!(uniset, "nfcinert", 1);
+    pub const NFDINERT_V1: DataKey = data_key!(uniset, "nfdinert", 1);
+    pub const NFKCINERT_V1: DataKey = data_key!(uniset, "nfkcinert", 1);
+    pub const NFKDINERT_V1: DataKey = data_key!(uniset, "nfkdinert", 1);
+    pub const PAT_SYN_V1: DataKey = data_key!(uniset, "Pat_Syn", 1);
+    pub const PAT_WS_V1: DataKey = data_key!(uniset, "Pat_WS", 1);
+    pub const PCM_V1: DataKey = data_key!(uniset, "PCM", 1);
+    pub const PRINT_V1: DataKey = data_key!(uniset, "print", 1);
+    pub const QMARK_V1: DataKey = data_key!(uniset, "QMark", 1);
+    pub const RADICAL_V1: DataKey = data_key!(uniset, "Radical", 1);
+    pub const RI_V1: DataKey = data_key!(uniset, "RI", 1);
+    pub const SD_V1: DataKey = data_key!(uniset, "SD", 1);
+    pub const SEGSTART_V1: DataKey = data_key!(uniset, "segstart", 1);
+    pub const SENSITIVE_V1: DataKey = data_key!(uniset, "Sensitive", 1);
+    pub const STERM_V1: DataKey = data_key!(uniset, "STerm", 1);
+    pub const TERM_V1: DataKey = data_key!(uniset, "Term", 1);
+    pub const UIDEO_V1: DataKey = data_key!(uniset, "UIdeo", 1);
+    pub const UPPER_V1: DataKey = data_key!(uniset, "Upper", 1);
+    pub const VS_V1: DataKey = data_key!(uniset, "VS", 1);
+    pub const WSPACE_V1: DataKey = data_key!(uniset, "WSpace", 1);
+    pub const XDIGIT_V1: DataKey = data_key!(uniset, "xdigit", 1);
+    pub const XIDC_V1: DataKey = data_key!(uniset, "XIDC", 1);
+    pub const XIDS_V1: DataKey = data_key!(uniset, "XIDS", 1);
+}
+
+/// Gets a locale-invariant default struct given a data key in this module's category.
+#[cfg(feature = "invariant")]
+pub(crate) fn get_invariant(data_key: &DataKey) -> Option<DataResponse<'static>> {
```

Comment:
minor:
Do we need a ref here if your only use of it is to just deref? In `get_prop` you grab the DataKey directly. Might be wrong here though